nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1513,6 +1515,63 @@ async fn disk_delete(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+#[derive(Display, Deserialize, JsonSchema)]
+#[display(style = "snake_case")]
+#[serde(rename_all = "snake_case")]
+pub enum DiskMetricName {
```

Comment:
It's a bit unfortunate that we have to duplicate these here. The interface to ClickHouse will [return an error](https://github.com/oxidecomputer/omicron/blob/b858ed729f1ae8393def124f20e76831cf20f442/oximeter/db/src/client.rs#L86) if the client selects a metric name that doesn't exist. We'd have to let the `metric_name` path component be a string. A "better" type is nice and all, but it really is a string. That'd at least let us evolve the metrics without changing this enum.

Reply:
Where's your concern regarding duplication - duplication between the HTTP API and the implementation in Crucible, which emits these metrics?

One perk of managing the "API type" independent of the "Clickhouse values" is that it allows us to add Crucible metrics into the system before exposing them through the public-facing API. However, it might be a matter of perspective on whether this is a good or bad thing...

Would you prefer we parse "any string" as input for the metric name, and let Clickhouse be the decision-maker regarding support?

Reply:
Yes, I'm asking about the duplication between the metrics as defined in Crucible and this enum. It just seems like an opportunity for drift, although I agree with your point that some decoupling allows independent evolution. And yes, an alternative would be accepting a string, letting ClickHouse client fail the query if there is no such metric.

I think this is probably fine for now. There are a number of details in this camp, the relationship between the raw data and the values we expose in the API, and this seems pretty reasonable at this point. Thanks!